app/pasteManager.js

```diff
@@ -0,0 +1,24 @@
+/* global MAXFILESIZE */
+import { bytes } from './utils';
+
+export default function(state, emitter) {
+  window.addEventListener('paste', event => {
+    if (state.route !== '/' || state.uploading) return;
+
+    for (const item of event.clipboardData.items) {
+      if (!item.type.includes('image')) continue;
+
+      const file = item.getAsFile();
+
+      if (!file) continue; // Sometimes null
+
+      if (file.size > MAXFILESIZE) {
+        // eslint-disable-next-line no-alert
+        alert(state.translate('fileTooBig', { size: bytes(MAXFILESIZE) }));
+        continue;
+      }
+
+      emitter.emit('upload', { file, type: 'paste' });
```

Comment:
We currently only allow one file to be uploaded at a time. How do you think we should handle that? One way would be to `return` here.

Reply:
@dannycoates Nice catch. Updated